### PR TITLE
fix(api): failure for flatpath

### PR DESCRIPTION
**Pull request by StarlightIbuki**
API definition with `flatpath` is not supposed to contain `parameters`, but my previous PR assumes it has.

### CODE REVIEW

**Comment by flrgh** [comment]
@StarlightIbuki can you check out the test failures? I guess we don't have busted tests wired into CI yet...

```
$ busted -o htest spec/02-request-build_spec.lua
======= Running tests from scanned files.
------- Global test environment setup.
------- Running tests from spec/02-request-build_spec.lua :
.......  RUN 21: unit: build_request simple flatPath
        __________
        FAILED  spec/02-request-build_spec.lua:21: unit: build_request simple flatPath
spec/02-request-build_spec.lua:26: Expected objects to be the same.
Passed in:
(string) 'https://storage.googleapis.com/test_service/test_service/p1/p2?p3'
Expected:
(string) 'test_service/p1/p2?p3'

stack traceback:
        spec/02-request-build_spec.lua:26: in function <spec/02-request-build_spec.lua:21>

   0.08 __________

.......  RUN 30: unit: build_request media check
        __________
        FAILED  spec/02-request-build_spec.lua:30: unit: build_request media check
spec/02-request-build_spec.lua:56: assertion failed!

stack traceback:
        spec/02-request-build_spec.lua:56: in function <spec/02-request-build_spec.lua:30>

   0.07 __________

------- 2 tests from spec/02-request-build_spec.lua (1.26 ms total)

------- Global test environment teardown.
======= 2 tests from 1 test file ran. (1.31 ms total)
PASSED  0 tests.

 FAILED 2 tests, listed below:
 FAILED spec/02-request-build_spec.lua:21: unit: build_request simple flatPath
 FAILED spec/02-request-build_spec.lua:30: unit: build_request media check

 2 FAILED TESTS
```

**Comment by StarlightIbuki** [comment]
I'm surprised that resty.gcp depends on resty.aws. Maybe we will address that in another PR.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          ~/.cache
+        key: ${{ runner.os }}-${{ matrix.openresty_version }}-cache
+
+    - name: Install Busted
+      run: |
+        /usr/local/openresty/luajit/bin/luarocks install busted
+
+    - uses: actions/checkout@v2
+
+    - name: Run tests
```
Comment (flrgh) [inline .github/workflows/tests.yml:50]: awesome, thanks for adding this @StarlightIbuki

**Review by flrgh — APPROVED** [review]

File: src/resty/gcp/init.lua [hunk 2]
```diff
@@ -79,14 +79,20 @@ local function build_request(accesstoken, apiDetail, baseUrl, params, requestBod
...
     end
 
-    local path = template_expansion(path_template, params)
+    local path = baseUrl .. template_expansion(path_template, params)
     local query
 
+    if apiDetail.flatPath then
+        return path, req
+    end
+
```
Comment (hanshuebner) [inline src/resty/gcp/init.lua:95]: @StarlightIbuki @flrgh This change prevents query parameters from being added to any APIs that have a `flatPath` defined, and most have.  It also prevents parameters from being validated.  I would like to remove this early return, but I'm wondering if we have enough tests to verify that things still work.  Any comments would be appreciated.

**Pull request merged**